### PR TITLE
Remove item(s) from  cart

### PR DESCRIPTION
**Pull request by ydaniju**
Why?
As a user, I should be able to remove items from my cart

How?
1. Remove items from localstorage
2. Update DOM afterwards

https://www.pivotaltracker.com/story/show/113278743
[Finishes #113278743]

### CODE REVIEW

**Comment by ChigboIO** [comment]
Great Job @andela-ydaniju. :+1: 

I found out that when I remove the first item in the cart dropdown, the total price resets to zero. However, this does not happen when I remove any other item.

I like the toast message when an item is removed. I think it will be nice to extend it to when an item is added. Oh! Wait!! That's not your feature, right? But you can still do it anyways. :wink: 

Good job, generally.

File: app/assets/javascripts/cart.js [hunk 1]
```diff
@@ -137,20 +125,57 @@ $(document).ready(function() {
         updateCart();
     });
 
+    $(".cart-span").on("click", ".del", function() {
+        var items = JSON.parse(localStorage.getmyshop_shoppping_cart_items);
+        var element_id = $(this).attr("id");
+        var cost = 0;
```
Comment (ChigboIO) [inline app/assets/javascripts/cart.js]: You might want to use one `var` for these

File: app/assets/javascripts/cart.js [hunk 2]
```diff
@@ -137,20 +125,57 @@ $(document).ready(function() {
...
+            location.reload();
+            Materialize.toast("Your cart has been emptied!", 3000);
+        } else {
+            for (var i in items) {
+                    if (items.hasOwnProperty(i)) {
+                    cost += (1 * items[i].quantity) * (1 * items[i].price);
+                    if (parseInt(element_id) == items[i].product_id) {
+                        var parent =
+                            document.getElementById("updated_product"),
+                            stringProductId = "" + items[i].product_id + "",
```
Comment (ChigboIO) [inline app/assets/javascripts/cart.js:145]: You can use the javascript `toString()` method here

Comment (ydaniju) [inline app/assets/javascripts/cart.js:145]: True @andela-echigbo, toString() was proving problematic. I found out you
need to wrap the number in parentheses to make it work. :)

On Mon, Feb 15, 2016 at 9:57 AM, Emmanuel Chigbo notifications@github.com
wrote:

> In app/assets/javascripts/cart.js
> https://github.com/andela/getmyshop/pull/44#discussion_r52872896:
> 
> > -    $(".cart-span").on("click", ".del", function() {
> > -        var items = JSON.parse(localStorage.getmyshop_shoppping_cart_items);
> > -        var element_id = $(this).attr("id");
> > -        var cost = 0;
> > -        if (items.length == 1) {
> > -            items[0] = undefined;
> > -            location.reload();
> > -            Materialize.toast("Your cart has been emptied!", 3000);
> > -        } else {
> > -            for (var i in items) {
> > -                    if (items.hasOwnProperty(i)) {
> > -                    cost += (1 \* items[i].quantity) \* (1 \* items[i].price);
> > -                    if (parseInt(element_id) == items[i].product_id) {
> > -                        var parent =
> > -                            document.getElementById("updated_product"),
> > -                            stringProductId = "" + items[i].product_id + "",
> 
> You can use the javascript toString() method here
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/andela/getmyshop/pull/44/files#r52872896.

## 

_Best Regards,_

_Yusuf Daniju /_\* Andela.com
http://andela.com/?utm_source=andela&utm_medium=email&utm_campaign=signature *
_Software Developer / *_+234.807.266.7248 <+234.807.266.7248>*\* / @Andela
http://bit.ly/andelatwitter*

*"The startup that's harder to get into than Harvard" - *CNN
http://snip.ly/HfId
See what The Wall Street Journal http://snip.ly/u9AI has to say.

**Comment by ChigboIO** [comment]
:+1:

**Comment by 0duaht** [comment]
:+1:

**Comment by oluosiname** [comment]
:+1:

**Comment by sofiengwin** [comment]
:+1:

**Pull request merged**